lib/etl/scrape.js

```diff
@@ -1,123 +1,82 @@
-    return Bluebird.mapSeries(timestampList, (item) => {
-        return api.MWcombatwzdate(object.gamer.username, item.start, item.end, object.gamer.platform).then((output) => {
-            let matches = output.matches || [];
-            sleep(7 * 1000)
-            console.log('Num matches for interval:' + matches.length);
-            return writeMatchesToDatabase(matches, gamer);
+function getQueryTimeframes(matchHistory, queriedTimestamps, isBackfill) {
+    let hasLatestTimestamp = queriedTimestamps.last_timestamp;
+    let matchesCopied = matchHistory
+    if (hasLatestTimestamp && !isBackfill) {
+        matchesCopied = matchHistory.filter((match)=>{
+            return  match.timestamp > parseFloat(queriedTimestamps.last_timestamp) * 1000
         });
-    });
-}
-
+    }
 
+    let timestampList = []
+    if(matchesCopied.length > 0){
+        timestampList = getTimestampList(matchesCopied, 'timestamp');
+    }
 
-/**
- * Takes a list of matches from the getMatchDetails API calls and writes them both
- * into the warzone.gamer_matches and warzone.matches tables
- * @param matches
- * @param gamer
- * @returns {PromiseLike<any> | Promise<any>}
- */
-function writeMatchesToDatabase(matches, gamer){
-    let mappedMatches = matches.map(WarzoneMapper.mapMatch);
-    let matchPromises = mappedMatches.map((m) => insertIntoDatabase(m, MATCH_TABLE));
-    let gamerMatches = matches.map((match) => WarzoneMapper.mapGamerMatch(match, gamer)).filter((match) => match.match_id && match.username);
-    return Bluebird.all(matchPromises).then(() => {
-        let gamerMatchPromises = gamerMatches.map((m) => insertIntoDatabase(m, GAMER_MATCH_TABLE));
-        return Bluebird.all(gamerMatchPromises);
-    });
+    return timestampList
 }
 
-
 /**
  *
  * Function that does the following:
  *
- * 1. Logs into the API
- * 2. Gets the match history for the gamer passed in
- * 3. Queries any necessary match details data to update
+ * 1. Gets the match history for the gamer passed in
+ * 2. Queries for the first and last timestamps already in the database
+ * 3. Filters the match history to only matches after the last timestamp in the database (or everything if IS_BACKFILL is true)
+ * 3. Queries any necessary match details data and inserts the data
  * 4. Finishes and returns.
  * @param gamer
+ * @param API
  * @param IS_BACKFILL
  * @returns {Promise<{gamer, matchHistory, queriedTimestamps, isBackfill}>}
  */
-function executePipeline(gamer, IS_BACKFILL = false) {
-    return API.login(EMAIL, PASSWORD)
-        .then(() => getMatchHistory(gamer, API, IS_BACKFILL)
-        .then((matches) => getMatchDetails(matches, API))
-        .then((data)=> {
-            console.log("done getting data for " + gamer.username + " on platform:" + gamer.platform);
-            return data;
-        })
-    );
+async function executePipeline(gamer, API, IS_BACKFILL = false) {
```

Comment:
Could we do some more descriptive naming? This just backfills data for gamers, right? something like findAndGenerateUnsavedGameData or generateGameDataForUser or something? What do you think?

Reply:
I like that. I'll update this to be `generateGameDataForUser`